crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -27,13 +27,14 @@ use web3::{ethabi::Token, types::CallRequest};
 /// A `TradeFinding`-based price estimator with request sharing and rate
 /// limiting.
 pub struct TradeEstimator {
-    inner: Inner,
+    inner: Arc<Inner>,
```

Comment:
What's the reason for moving the `Inner` into an `Arc`?
I could only come up with slightly better performance or future proofing the `Inner`'s  `Clone`-ing capabilities against possible changes.
Not sure if those are worth the change at the moment. No strong opinion, though.

Reply:
> I could only come up with slightly better performance or future proofing the Inner's Clone-ing capabilities against possible changes.

I just thought it made more sense for `Inner` to be clone, so you don't have to think about adding large structures to it in the future.

I thought the code changes weren't that bad (but can split it out into a separate PR if you prefer).

Reply:
I think the change is small enough to be included in this PR. Was mostly curious if I overlooked some major advantage.

Reply:
It was more about "semantically we want to share `Inner` across futures".